ambry-store/src/test/java/com.github.ambry.store/CuratedLogIndexState.java

```diff
@@ -391,6 +420,16 @@ void verifyEntriesForHardDeletes(Set<MockId> deletedIds) throws IOException {
     }
   }
 
+  boolean shouldCountAsDeletedAt(MockId id, long referenceTimeMs) {
```

Comment:
All implementation in CuratedLogIndexState is provided by gopal, including additions requested by me. There are some updates in CuratedLogIndexState which i will push to here along with the commit to address other comments.